beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -205,11 +205,7 @@ func matchingAtts(atts []ethpbalpha.Att, slot primitives.Slot, attDataRoot []byt
 		// compare the committee index separately.
 		if postElectra {
 			if att.Version() >= version.Electra {
```

Comment:
The whole block reads better as follows

```
        root, err := att.GetData().HashTreeRoot()
		if err != nil {
			return nil, errors.Wrap(err, "could not get attestation data root")
		}
		if !bytes.Equal(root[:], attDataRoot) {
			continue
		}
        if att.Version() < version.Electra || (postElectra && att.GetCommitteeIndex()  == index) {
			result = append(result, att)
        }
```

Reply:
Agreed, but we need to have `(!postElectra && att.Version() < version.Electra) || ...`

Reply:
that should be covered for att.Version() < version.Electra since we should clean the pool at the fork, but anyway you can change the first side of the `||` to include the check
```
        if (!postElectra && att.Version() < version.Electra) || (postElectra && att.GetCommitteeIndex()  == index) {
```